test/unit/testCheckAnswersSummaryJSONObjectBuilder.js

```diff
@@ -0,0 +1,148 @@
+    beforeEach(() => {
+        checkAnswersSummaryJSONObjBuilder = new CheckAnswersSummaryJSONObjectBuilder();
+    });
+
+    describe('build()', () => {
+        it('should build the json object from html', (done) => {
+            const checkAnswersSummary = checkAnswersSummaryJSONObjBuilder.build(html);
+            assert.exists(checkAnswersSummary);
+            assertPropertyExistsAndIsEqualTo(checkAnswersSummary.mainParagraph,
+                'Check the information below carefully. This will form a record of your application for probate. It will also be stored as a public record, and will be able to be viewed online.');
+            assertPropertyExistsAndIsEqualTo(checkAnswersSummary.pageTitle, '\n' +
+                '        Check your answers\n' +
+                '    ');
+            assert.isArray(checkAnswersSummary.sections, 'Sections exists');
+            assert.lengthOf(checkAnswersSummary.sections, 5, 'Section array has length of 5');
+
+            const willSection = checkAnswersSummary.sections[0];
+            assertPropertyExistsAndIsEqualTo(willSection.title, '\n' +
+                '        The will\n' +
+                '    ');
+            assertPropertyExistsAndIsEqualTo(willSection.type, 'heading-medium');
+            assert.isArray(willSection.questionAndAnswers);
+            assert.lengthOf(willSection.questionAndAnswers, 2, 'Will Section array has 2 questionsAndAnswers');
+
+            assertQuestionAndAnswer(willSection.questionAndAnswers[0], 'Did the person who died leave a will?', 'Yes');
+            assertQuestionAndAnswer(willSection.questionAndAnswers[1], 'Do you have the original will?', 'Yes');
+
+            const ihtSection = checkAnswersSummary.sections[1];
+            assertPropertyExistsAndIsEqualTo(ihtSection.title, 'Inheritance tax');
+            assertPropertyExistsAndIsEqualTo(ihtSection.type, 'heading-medium');
+
+            assert.isArray(ihtSection.questionAndAnswers);
+            assert.lengthOf(ihtSection.questionAndAnswers, 2, 'IHT Section array has 2 questionsAndAnswers');
+
+            assertQuestionAndAnswer(ihtSection.questionAndAnswers[0], 'Has an Inheritance Tax (IHT) form been filled in?', 'Yes');
+            assertQuestionAndAnswer(ihtSection.questionAndAnswers[1], 'How was the Inheritance Tax (IHT) form submitted?', 'By post');
+
+            const executorsSection = checkAnswersSummary.sections[2];
+            assertPropertyExistsAndIsEqualTo(executorsSection.title, 'The executors');
+            assertPropertyExistsAndIsEqualTo(executorsSection.type, 'heading-medium');
+
+            assert.isArray(executorsSection.questionAndAnswers);
+            assert.lengthOf(executorsSection.questionAndAnswers, 1, 'Executors Section array has 1 questionsAndAnswers');
+            assertQuestionAndAnswer(executorsSection.questionAndAnswers[0], 'How many past and present executors are named on the will and any updates (‘codicils’)?', '1');
```

Comment:
Not sure about this, but... `&lsquo;` and `&rsquo;`?

Reply:
i couldn't get this to work, i dont think that the html escape values are working in this conext.